### ISSUE TITLE
Split application into 4 separate applications

### ISSUE THREAD

**Issue by tomfitzhenry** [certificate-transparency-watch/hs-certificate-transparency#35]
hs-certificate-transparency is 4 repeatedly created threads that share source code and a DB, but don't pass state amongst themselves. It might be sensible to split these into 4 processes.

This would have the advantage of:
* each process could be started/stopped independently
* could remove the ugly code I have that does the repeated thread creation, including error handing
* easily reconfigure the frequency at which the processes start, if I started them with cron or systemd.timer, without having to recompile
* per process cgroups
* per process capabilities

Basically I could use this as an excuse to learn about systemd!

This is where the 4 threads are created in code: https://github.com/certificate-transparency-watch/hs-certificate-transparency/blob/master/Network/CertificateTransparency/Main.hs#L41-L44
